pkg/reconciler/domainmapping/reconciler.go

```diff
@@ -183,6 +183,23 @@ func certClass(ctx context.Context) string {
 }
 
 func (r *Reconciler) tls(ctx context.Context, dm *v1alpha1.DomainMapping) ([]netv1alpha1.IngressTLS, []netv1alpha1.HTTP01Challenge, error) {
+	if dm.Spec.TLS != nil {
+		namespace := dm.Namespace
+		if dm.Spec.TLS.SecretNamespace != "" {
+			namespace = dm.Spec.TLS.SecretNamespace
+		}
+		// Assume the provided certificate is ready
+		dm.Status.MarkCertificateReady("")
```

Comment:
nit: Does this need to happen ? As a Kcert isn't created in this flow, ideally we don't have to set any statuses corresponding to it 🤔 

This func sets apis.ConditionType = "CertificateProvisioned", which isn't literally what is happening.

Reply:
Yeah, I get what you mean but the Cert condition is a dependency for the DomainMapping ready condition, I think it should be possible to have a different set of conditions in the case you provide an existing TLS certificate so I'll check into that

Reply:
We do have the ability to mark that the cert provisioning condition is true without indicating that it was actually provisioned with the [`MarkTLSNotEnabled`](https://github.com/knative/serving/blob/main/pkg/apis/serving/v1alpha1/domainmapping_lifecycle.go#L70) method, would something like that work here?